app/src/main/java/com/pps/globant/fittracker/utils/MyAsyncTask.java

```diff
@@ -0,0 +1,83 @@
+package com.pps.globant.fittracker.utils;
+
+import android.content.SharedPreferences;
+import android.os.AsyncTask;
+
+import com.pps.globant.fittracker.mvp.model.InstagramLoginModel;
+import com.squareup.otto.Bus;
+
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.OutputStreamWriter;
+import java.net.URL;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import static com.pps.globant.fittracker.utils.CONSTANTS.CLIENT_ID;
+import static com.pps.globant.fittracker.utils.CONSTANTS.CLIENT_SECRET;
+import static com.pps.globant.fittracker.utils.CONSTANTS.REDIRECT_URI;
+import static com.pps.globant.fittracker.utils.CONSTANTS.SP_CODE;
+import static com.pps.globant.fittracker.utils.CONSTANTS.SP_NAME;
+import static com.pps.globant.fittracker.utils.CONSTANTS.SP_TOKEN;
+
+public class MyAsyncTask extends AsyncTask<URL, Integer, Long> {
```

Comment:
parameter to instance the AsyncTask<URL, Integer, Long are wrong. you don't need a Long as result, you didn't ever use it. same to the Integer second paramer for onProgress () method. instead of saving a private variable for the full name return it in doInBackground and accesing it in onPostExecute.

Reply:
ok! thanks